doc/contributing/coding-conventions.chapter.md

```diff
@@ -453,6 +453,9 @@ In the file `pkgs/top-level/all-packages.nix` you can find fetch helpers, these
   }
   ```
 
+When fetching from GitHub, commits must always be referenced by their full commit hash. This is because GitHub shares commit hashes between all forks and returns 404 if a short commit hash is ambiguous. For some 6-character commit hashes in `nixpkgs` this is already the case.
```

Comment:
English is not my native tongue, however I suggest:
```suggestion
When fetching from GitHub, commits must always be referenced by their full commit hash. This is because GitHub shares commit hashes among all forks and returns 404 when a short commit hash is ambiguous. It already happened for some short, 6-character commit hashes in `nixpkgs`.
```